server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/CountValueAggregator.java

```diff
@@ -7,22 +7,22 @@
  */
 package org.opensearch.index.compositeindex.datacube.startree.aggregators;
 
-import org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype.StarTreeNumericType;
+import org.opensearch.index.mapper.NumberFieldMapper;
 
 /**
  * Count value aggregator for star tree
  *
  * @opensearch.experimental
  */
-class CountValueAggregator implements ValueAggregator<Long> {
+public class CountValueAggregator implements ValueAggregator<Long> {
 
     public static final long DEFAULT_INITIAL_VALUE = 1L;
-    private static final StarTreeNumericType VALUE_AGGREGATOR_TYPE = StarTreeNumericType.LONG;
+    private static final NumberFieldMapper.NumberType VALUE_AGGREGATOR_TYPE = NumberFieldMapper.NumberType.LONG;
```

Comment:
Lets make it type of converter and let the converter be generic as well instead of tying to numeric fields